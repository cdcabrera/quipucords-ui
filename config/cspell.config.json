{
  "language": "en",
  "words": [
    "adipiscing",
    "amet",
    "axios",
    "bezier",
    "bgimages",
    "brms",
    "creds",
    "crosshairs",
    "describedby",
    "elit",
    "fadein",
    "generatedid",
    "labelledby",
    "mturley",
    "nocolor",
    "novalidate",
    "ouia",
    "paramiko",
    "patternfly",
    "pficon",
    "progressbar",
    "quipucords",
    "rescan",
    "rhacs",
    "rowgroup",
    "scanjob",
    "tabindex",
    "tbody",
    "thead",
    "touchspin",
    "vals",
    "valuetext"
  ]
}
