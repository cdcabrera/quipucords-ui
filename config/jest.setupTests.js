import { act } from 'react';
import { render, renderHook } from '@testing-library/react';
import { dotenv } from 'weldable';

/**
 * Set dotenv params.
 */
dotenv.setupDotenvFilesForEnv({ env: process.env.NODE_ENV });

/**
 * Emulate i18next setup
 */
jest.mock('i18next', () => {
  const Test = function () { // eslint-disable-line
    this.use = () => this;
    this.init = () => Promise.resolve();
    this.changeLanguage = () => Promise.resolve();
  };
  return new Test();
});

/**
 * Emulate for translation hook for snapshots
 */
jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({ t: (...args) => args })
}));

/**
 * Quick React function component and hook results testing. Based off of classic Enzyme testing,
 * Use "shallowComponent" if
 * - the component is a function, class results may not be expected
 * - you want a quick component response typically determined by a condition
 * - snapshot size needs to be reduced
 * - the insanity of having a more complicated testing API than actual React components/hooks has finally gotten to you
 *
 * @param {React.ReactNode} testComponent
 * @returns {{unmount: () => void, render: () => any, getHTML: () => any, querySelector: () => any,
 *     querySelectorAll: () => any, setProps: () => Promise<any> } | any}
 */
export const shallowComponent = async testComponent => {
  const localRenderHook = async (component, updatedProps) => {
    if (typeof component?.type === 'function') {
      try {
        let result;
        let unmount;

        // const { result: reactTestingResult, rerender, unmount: reactTestingUnmount } = testingRenderHook(() =>
        const { result: reactTestingResult, unmount: reactTestingUnmount } = renderHook(() =>
          component.type({ ...component.props, ...updatedProps })
        );
        await act(async () => {
          result = await reactTestingResult;
        });

        unmount = reactTestingUnmount;

        if (!result || !result.current || typeof result.current === 'string' || typeof result.current === 'number') {
          return result.current;
        }

        if (typeof result.current === 'function') {
          return result.current.toString();
        }

        const getHTML = (sel, _internalRender = result.current) => {
          const { container } = render(_internalRender);
          return container.innerHTML;
        };

        const renderComponent = (sel = '*', _internalRender = result.current) => {
          if (!_internalRender || typeof _internalRender === 'string' || typeof _internalRender === 'number') {
            return _internalRender;
          }

          if (typeof _internalRender === 'function') {
            return _internalRender.toString();
          }

          const { container } = render(_internalRender);
          return container.querySelector(sel);
        };

        const querySelector = (sel, _internalRender = result.current) => {
          const { container } = render(_internalRender);
          return container.querySelector(sel);
        };

        const querySelectorAll = (sel, _internalRender = result.current) => {
          const { container } = render(_internalRender);
          return container.querySelectorAll(sel);
        };

        const setProps = async p => localRenderHook(component, p);

        if (Array.isArray(result.current)) {
          const updatedR = result.current;
          updatedR.unmount = unmount;
          updatedR.render = () => result.current.map(res => renderComponent(undefined, res));
          updatedR.getHTML = getHTML;
          updatedR.querySelector = querySelector;
          updatedR.querySelectorAll = querySelectorAll;
          updatedR.setProps = setProps;
          return updatedR;
        }

        return {
          ...result.current,
          unmount,
          // rerender,
          render: renderComponent,
          find: querySelector,
          getHTML,
          querySelector,
          querySelectorAll,
          setProps
        };
      } catch (e) {
        console.error(e.message);
      }
    }

    return component;
  };

  return localRenderHook(testComponent);
};

export const shallowHook = shallowComponent;
