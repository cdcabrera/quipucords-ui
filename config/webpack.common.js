const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const HtmlReplaceWebpackPlugin = require('html-replace-webpack-plugin');
const CopyPlugin = require('copy-webpack-plugin');
const { setupWebpackDotenvFilesForEnv, setupDotenvFilesForEnv } = require('./build.dotenv');

const {
  _BUILD_BG_IMAGES_DIRNAME: BG_IMAGES_DIRNAME,
  _BUILD_DIST_DIR: DIST_DIR,
  _BUILD_FAVICON: FAVICON,
  _BUILD_OUTPUT_ONLY: OUTPUT_ONLY,
  _BUILD_PUBLIC_PATH: PUBLIC_PATH,
  _BUILD_RELATIVE_DIRNAME: RELATIVE_DIRNAME,
  _BUILD_SRC_DIR: SRC_DIR,
  _BUILD_STATIC_DIR: STATIC_DIR,
  _BUILD_UI_NAME: UI_NAME
} = setupDotenvFilesForEnv({ env: process.env.NODE_ENV });

if (OUTPUT_ONLY !== true) {
  console.info(
    `\nPrepping files...\n  SRC DIR: ${SRC_DIR}\n  OUTPUT DIR: ${DIST_DIR}\n  PUBLIC PATH: ${PUBLIC_PATH}\n`
  );
}

module.exports = env => ({
  entry: {
    app: path.join(SRC_DIR, 'index.tsx')
  },
  module: {
    rules: [
      {
        test: /\.(tsx|ts|jsx|js)?$/,
        include: [SRC_DIR],
        use: [
          {
            loader: 'babel-loader',
            options: {
              transpileOnly: env !== 'development',
              experimentalWatchApi: true
            }
          }
        ]
      },
      {
        test: /\.(svg|ttf|eot|woff|woff2)$/,
        // only process modules with this loader
        // if they live under a 'fonts' or 'pficon' directory
        include: [
          path.resolve(RELATIVE_DIRNAME, 'node_modules/patternfly/dist/fonts'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/react-core/dist/styles/assets/fonts'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/react-core/dist/styles/assets/pficon'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/patternfly/assets/fonts'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/patternfly/assets/pficon')
        ],
        use: {
          loader: 'file-loader',
          options: {
            // Limit at 50k. larger files emitted into separate files
            limit: 5000,
            outputPath: 'fonts',
            name: '[name].[ext]'
          }
        }
      },
      {
        test: /\.svg$/,
        include: input => input.indexOf('background-filter.svg') > 1,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'svgs',
              name: '[name].[ext]'
            }
          }
        ]
      },
      (() => {
        if (BG_IMAGES_DIRNAME) {
          return {
            test: /\.svg$/,
            // only process SVG modules with this loader if they live under a 'bgimages' directory
            // this is primarily useful when applying a CSS background using an SVG
            include: input => input.indexOf(BG_IMAGES_DIRNAME) > -1,
            use: {
              loader: 'svg-url-loader',
              options: {
                limit: 10000
              }
            }
          };
        }
      })(),
      {
        test: /\.svg$/,
        // only process SVG modules with this loader when they don't live under a 'bgimages',
        // 'fonts', or 'pficon' directory, those are handled with other loaders
        include: input =>
          input.indexOf(BG_IMAGES_DIRNAME) === -1 &&
          input.indexOf('fonts') === -1 &&
          input.indexOf('background-filter') === -1 &&
          input.indexOf('pficon') === -1,
        use: {
          loader: 'raw-loader',
          options: {}
        }
      },
      {
        test: /\.(jpg|jpeg|png|gif)$/i,
        include: [
          SRC_DIR,
          path.resolve(RELATIVE_DIRNAME, 'node_modules/patternfly'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/patternfly/assets/images'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/react-styles/css/assets/images'),
          path.resolve(RELATIVE_DIRNAME, 'node_modules/@patternfly/react-core/dist/styles/assets/images'),
          path.resolve(
            RELATIVE_DIRNAME,
            'node_modules/@patternfly/react-core/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            RELATIVE_DIRNAME,
            'node_modules/@patternfly/react-table/node_modules/@patternfly/react-styles/css/assets/images'
          ),
          path.resolve(
            RELATIVE_DIRNAME,
            'node_modules/@patternfly/react-inline-edit-extension/node_modules/@patternfly/react-styles/css/assets/images'
          )
        ],
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 5000,
              outputPath: 'images',
              name: '[name].[ext]'
            }
          }
        ]
      },
      {
        test: /\.s[ac]ss$/i,
        use: [
          // Creates `style` nodes from JS strings
          'style-loader',
          // Translates CSS into CommonJS
          'css-loader',
          // Compiles Sass to CSS
          'sass-loader'
        ]
      },
      {
        test: /\.ya?ml$/,
        use: 'js-yaml-loader'
      }
    ]
  },
  output: {
    filename: '[name].bundle.js',
    path: DIST_DIR,
    publicPath: PUBLIC_PATH
  },
  plugins: [
    ...setupWebpackDotenvFilesForEnv({
      directory: RELATIVE_DIRNAME,
      isRoot: true
    }),
    new HtmlWebpackPlugin({
      template: path.join(SRC_DIR, 'index.html'),
      ...(UI_NAME && { title: UI_NAME }),
      ...(FAVICON && { favicon: path.resolve(RELATIVE_DIRNAME, FAVICON) })
    }),
    new HtmlReplaceWebpackPlugin([
      [
        {
          pattern: /%([A-Z_]+)%/g,
          replacement: (match, $1) => process.env?.[$1] || match
        }
      ]
    ]),
    new CopyPlugin({
      patterns: [
        {
          from: path.join(STATIC_DIR, 'locales'),
          to: path.join(DIST_DIR, 'locales'),
          noErrorOnMissing: true
        },
        {
          from: path.join(STATIC_DIR, 'images'),
          to: path.join(DIST_DIR, 'images'),
          noErrorOnMissing: true
        },
        {
          from: path.join(SRC_DIR, 'images'),
          to: path.join(DIST_DIR, 'images'),
          noErrorOnMissing: true
        },
        {
          from: path.join(STATIC_DIR, 'favicon.ico'),
          to: path.join(DIST_DIR),
          noErrorOnMissing: true
        },
        {
          from: path.join(STATIC_DIR, 'favicon.png'),
          to: path.join(DIST_DIR),
          noErrorOnMissing: true
        },
        {
          from: path.join(STATIC_DIR, 'manifest.json'),
          to: path.join(DIST_DIR),
          noErrorOnMissing: true
        },
        {
          from: path.join(STATIC_DIR, 'robots.txt'),
          to: path.join(DIST_DIR),
          noErrorOnMissing: true
        }
      ]
    })
  ],
  resolve: {
    extensions: ['.js', '.ts', '.tsx', '.jsx'],
    alias: {
      '~': path.resolve(SRC_DIR)
    },
    symlinks: false,
    cacheWithContext: false
  }
});
