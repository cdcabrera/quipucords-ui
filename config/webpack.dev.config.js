const {
  // cssLoaderResolve,
  // sassLoaderResolve,
  EslintWebpackPlugin
  // MiniCssExtractPlugin
} = require('weldable/lib/packages');

module.exports = ({ SRC_DIR } = {}) => ({
  /*
  module: {
    rules: [
      {
        test: /\.(sa|sc)ss$/i,
        use: [MiniCssExtractPlugin.loader, cssLoaderResolve, sassLoaderResolve]
      }
    ]
  },
  */
  plugins: [
    new EslintWebpackPlugin({
      context: SRC_DIR,
      failOnError: false
    })
  ]
});
