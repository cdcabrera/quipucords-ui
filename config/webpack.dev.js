// const path = require('path');
const { merge } = require('webpack-merge');
const ESLintPlugin = require('eslint-webpack-plugin');
const { setupWebpackDotenvFilesForEnv, setupDotenvFilesForEnv } = require('./build.dotenv');

const {
  _BUILD_DIST_DIR: DIST_DIR,
  _BUILD_HOST: HOST,
  _BUILD_RELATIVE_DIRNAME: RELATIVE_DIRNAME,
  _BUILD_PORT: PORT,
  _BUILD_SRC: SRC_DIR
} = setupDotenvFilesForEnv({ env: 'development' });

const webpackCommon = require('./webpack.common');

const devWebpack = merge(
  {
    plugins: [
      ...setupWebpackDotenvFilesForEnv({
        directory: RELATIVE_DIRNAME,
        env: 'development'
      }),
      new ESLintPlugin({
        context: SRC_DIR,
        failOnError: false
      })
    ]
  },
  webpackCommon('development'),
  {
    mode: 'development',
    devtool: 'eval-source-map',
    devServer: {
      host: HOST,
      port: PORT,
      compress: true,
      historyApiFallback: true,
      hot: true,
      open: true,
      devMiddleware: {
        stats: 'errors-only',
        writeToDisk: true
      },
      client: {
        overlay: false,
        progress: false
      },
      static: {
        directory: DIST_DIR
      },
      watchFiles: {
        paths: ['src/**/*', 'public/**/*']
      }
    },
    module: {}
  }
);

console.log(devWebpack);

module.exports = devWebpack;
