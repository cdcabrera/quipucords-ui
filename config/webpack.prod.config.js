// const {
//  sassLoaderResolve,
//  MiniCssExtractPlugin
// } = require('weldable/lib/packages');

module.exports = () => ({
  ignoreWarnings: [
    {
      message: /mini-css-extract-plugin/
    }
  ]
  /*
  module: {
    rules: [
      {
        test: /\.(sa|sc)ss$/i,
        use: [sassLoaderResolve]
      }
    ]
  }
  */
  /*
  plugins: [
    new MiniCssExtractPlugin({
      chunkFilename: '[name].[contenthash:8].chunk.css',
      filename: '[name].[contenthash:8].css',
      ignoreOrder: true
    })
  ]
  */
});
