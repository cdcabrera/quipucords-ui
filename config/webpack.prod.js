const fs = require('fs');
const path = require('path');

console.log('>>>>>>>>>>>>>>>>>>> WORK');

const updateTsConfig = ({ filePath, addConfiguration, encoding = 'utf8' } = {}) => {
  const currentConfig = require(filePath);
  // const updatedContents = JSON.stringify({ ...currentConfig, ...addConfiguration }, null, 2);



  console.log('>>>>>>>>>>>>>>>>>>>>>>>');
  // console.log(JSON.parse(JSON.stringify(currentConfig)));
  // console.log();
  // console.log(updatedContents);
  console.log('>>>>>>>>>>>>>>>>>>>>>>>');
  // console.log(undefined);
  console.log('>>>>>>>>>>>>>>>>>>>>>>>');

  try {
    // const currentConfig = JSON.parse(fs.readFileSync(filePath, { encoding }));
    const updatedContents = JSON.stringify({ ...currentConfig, ...addConfiguration }, null, 2);
    fs.writeFileSync(filePath, updatedContents, { encoding });
  } catch (e) {
    console.error(e.message);
  }
};

module.exports = ({ RELATIVE_DIRNAME } = {}) => {
  updateTsConfig({
    filePath: path.resolve(process.cwd(), 'tsconfig.json'),
    addConfiguration: {
      exclude: [
        '**/*.test.ts',
        '**/*.test.tsx'
      ]
    }
  });

  return {
    ignoreWarnings: [
      {
        message: /mini-css-extract-plugin/
      }
    ]
  };
}
