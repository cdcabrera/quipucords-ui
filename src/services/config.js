import axios from 'axios';
import cookies from 'js-cookie';

/**
 * Set Axios XHR default timeout.
 */
const globalXhrTimeout = Number.parseInt(process.env.REACT_APP_AJAX_TIMEOUT, 10) || 60000;

/**
 * Return a formatted auth header.
 *
 * @returns {{}}
 */
const authHeader = () => {
  const authToken = cookies.get(process.env.REACT_APP_AUTH_TOKEN) || '';

  if (authToken === '') {
    return {};
  }

  return {
    [process.env.REACT_APP_AUTH_HEADER]: (process.env.REACT_APP_AUTH_HEADER_CONTENT || '{0}').replace('{0}', authToken)
  };
};

/**
 * Apply custom service config.
 *
 * @param {object} passedConfig
 * @param {object} options
 * @param {boolean} options.auth
 * @param {number} options.xhrTimeout
 * @returns {{headers: *}}
 */
const serviceConfig = (passedConfig = {}, { auth = true, xhrTimeout = globalXhrTimeout } = {}) => ({
  headers: auth ? authHeader() : {},
  timeout: xhrTimeout,
  ...passedConfig
});

/**
 * Use a global Axios configuration.
 *
 * @param {object} config
 * @param {object} options
 * @returns {Promise<*>}
 */
const serviceCall = async (config, options) => axios(serviceConfig(config, options));

const config = { serviceCall, globalXhrTimeout, serviceConfig, authHeader };

export { config as default, config, authHeader, globalXhrTimeout, serviceCall, serviceConfig };
