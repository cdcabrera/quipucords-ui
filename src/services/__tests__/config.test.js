import cookies from 'js-cookie';
import { config } from '../config';

describe('ServiceConfig', () => {
  it('should have specific properties and methods', () => {
    expect(Object.keys(config)).toMatchSnapshot('specific props and methods');
  });

  it('should export a default services config', () => {
    expect(config.serviceConfig).toBeDefined();

    cookies.set(process.env.REACT_APP_AUTH_TOKEN, 'spoof');

    const configObject = config.serviceConfig(
      {
        method: 'post',
        timeout: 3
      },
      true
    );

    expect(configObject.method).toEqual('post');
    expect(configObject.timeout).toEqual(3);
  });

  it('should export a default services config without authorization', () => {
    const configObject = config.serviceConfig({}, { auth: false });

    expect(configObject.headers[process.env.REACT_APP_AUTH_HEADER]).toBeUndefined();
  });
});
