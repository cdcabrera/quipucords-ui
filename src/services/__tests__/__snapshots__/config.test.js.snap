// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceConfig should have specific properties and methods: specific props and methods 1`] = `
Array [
  "serviceCall",
  "globalXhrTimeout",
  "serviceConfig",
  "authHeader",
]
`;
